### PR TITLE
tests: move net related applications to test/net

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

As suggested on Matrix, this PR moves all network related test applications to `tests/net`.

### Testing procedure

- Green CI (e.g once all broken symlinks are fixed...)

### Issues/PRs references

#19551 and the like

### CODE REVIEW

File: tests/net/gnrc_rpl/tests/01-run.py [hunk 1]
```diff
@@ -162,8 +165,7 @@ def test_alternative_route(factory, zep_dispatch):
 
 
 def run_test(func, factory):
-    with Popen(['../../dist/tools/zep_dispatch/bin/zep_dispatch',
-                '-t', '-', '127.0.0.1', '17754'], stdin=subprocess.PIPE) as zep_dispatch:
+    with Popen([ZEP_DISPATCH_PATH, '-t', '-', '127.0.0.1', '17754'], stdin=subprocess.PIPE) as zep_dispatch:
```
Comment (maribu) [inline tests/net/gnrc_rpl/tests/01-run.py:168]: nice cleanup :)

**Review by maribu — APPROVED** [review]
ACK, thx

File: tests/net/gnrc_rpl/tests/01-run.py (hunk 1 above)
Comment (aabadie) [reply tests/net/gnrc_rpl/tests/01-run.py:168]: Hmm, it seems to be required in other places...

**Comment by aabadie** [comment]
bors merge

**Pull request merged**